api/lib/domain/models/CampaignParticipant.js

```diff
@@ -69,7 +69,12 @@ class CampaignParticipant {
         `User ${this.userIdentity.id} has already a campaign participation with campaign ${this.campaignToStartParticipation.id}`
       );
     }
-    if (this.previousCampaignParticipation && this.previousCampaignParticipation.status !== 'SHARED') {
+
+    if (this.previousCampaignParticipation?.isDeleted) {
+      throw new ForbiddenAccess(couldNotImproveCampaignErrorMessage);
+    }
+
+    if (this.previousCampaignParticipation && this.previousCampaignParticipation?.status !== 'SHARED') {
```

Comment:
suggestion: On a besoin du this.previousCampaignParticipation ?

Reply:
yes, si on l'a pas ça casse les tests, parce que si on a pas de `previousCampaignParticipation` alors on rentre dans le if, parce que la condition est négative. Mais sinon j'ai trouvé une autre manière de l'écrire et là ça passe les tests, wdyt ?
`['STARTED', 'TO_SHARE'].includes(this.previousCampaignParticipation?.status)`